windows/test-app.js

```diff
@@ -94,6 +94,122 @@ function findUserProjects(projectDir, projects = []) {
   }, projects);
 }
 
+/**
+ * Finds NuGet dependencies.
+ *
+ * Visual Studio (?) currently does not download transitive dependencies. This
+ * is a workaround until `react-native-windows` autolinking adds support.
+ *
+ * @returns {[string, string][]}
+ */
+function getNuGetDependencies() {
+  const pkgJson = findNearest("package.json");
+  if (!pkgJson) {
+    return [];
+  }
+
+  const { loadConfig } = require("@react-native-community/cli");
+  const dependencies = Object.values(loadConfig().dependencies);
+
+  const { XMLParser } = require("fast-xml-parser");
+  const xml = new XMLParser({
+    ignoreAttributes: false,
+    transformTagName: (tag) => tag.toLowerCase(),
+  });
+
+  const lowerCase = (/** @type{Record<string, string>} */ refs) => {
+    for (const key of Object.keys(refs)) {
+      refs[key.toLowerCase()] = refs[key];
+    }
+    return refs;
+  };
+
+  /** @type {Record<string, [string, string]>} */
+  const packageRefs = {};
+
+  for (const { root, platforms } of dependencies) {
+    /** @type {{ projects?: Record<string, string>[]; sourceDir?: string; }?} */
+    const windows = platforms?.["windows"];
+    if (!windows || !Array.isArray(windows.projects)) {
+      continue;
+    }
+
+    const projects = windows.projects.map(({ projectFile }) =>
+      path.join(root, windows.sourceDir || ".", projectFile)
+    );
+
+    if (!Array.isArray(projects)) {
+      continue;
+    }
+
+    // Look for `PackageReference` entries:
+    //
+    //     <Project>
+    //         <ImportGroup>
+    //             <PackageReference ... />
+    //             <PackageReference ... />
+    //         </ImportGroup>
+    //     </Project>
+    //
+    for (const vcxproj of projects) {
+      const proj = xml.parse(fs.readFileSync(vcxproj, textFileReadOptions));
+      const itemGroup = proj.project?.itemgroup;
+      if (!itemGroup) {
+        continue;
+      }
+
+      const itemGroups = Array.isArray(itemGroup) ? itemGroup : [itemGroup];
+      for (const group of itemGroups) {
+        const pkgRef = group["packagereference"];
+        if (!pkgRef) {
+          continue;
+        }
+
+        const refs = Array.isArray(pkgRef) ? pkgRef : [pkgRef];
+        for (const ref of refs) {
+          // Attributes are not case-sensitive
+          lowerCase(ref);
+
+          const id = ref["@_include"];
+          const version = ref["@_version"];
+          if (!id || !version) {
+            continue;
+          }
+
+          // Package ids are not case-sensitive
+          packageRefs[id.toLowerCase()] = [id, version];
+        }
+      }
+    }
+  }
+
+  // Remove dependencies managed by us
+  [
+    "microsoft.reactnative",
+    "microsoft.reactnative.cxx",
+    "microsoft.ui.xaml",
+    "microsoft.windows.cppwinrt",
+    "reactnative.hermes.windows",
+    "nlohmann.json",
```

Comment:
any way we can have this list not hardcoded/manually maintained? How likely is it to change?